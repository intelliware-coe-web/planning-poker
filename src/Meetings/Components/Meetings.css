.meeting-card, .meeting-button {
    background: #2359fa;
    background-color: #2359fa;
}

.uk-card-title {
    font-family: 'Mandali', sans-serif;
}

.add-card {
    background: #c702a9;
}

.meeting-card:hover, .uk-button-secondary:hover, .add-card:hover {
    background-color: #b1bafc;
}

a:hover {
    text-decoration: none;
}