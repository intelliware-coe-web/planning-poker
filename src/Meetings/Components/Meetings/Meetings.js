import React from 'react';
import MeetingButtons from '../MeetingButtons/MeetingButtons'
import AddMeetingButton from '../AddMeetingButton/AddMeetingButton'

export default function Meetings() {

    return (
        <div className="default-background">
            <div className="uk-container uk-text-center@m">
                <h1 className="uk-heading-divider uk-margin-top">Meetings</h1>
                <div className="uk-align-center uk-width-1-2@m">                
                    <MeetingButtons />
                    <AddMeetingButton />
                </div>
            </div>
        </div>
    );
}
