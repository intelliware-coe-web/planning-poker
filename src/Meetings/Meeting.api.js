import {getData, postData} from '../Fetch.api';

export const MeetingAPI = {
  all: () => getData(`https://planning-poker-241613.appspot.com/meetings`),
  byId: meetingId => getData(`https://planning-poker-241613.appspot.com/meetings/${meetingId}`),
  create: (body) => postData(`https://planning-poker-241613.appspot.com/meetings`, body),
  addAttendee: (meetingId, body) => postData(`https://planning-poker-241613.appspot.com/meetings/${meetingId}/attendees`, body),
  addTicket: (meetingId, body) => postData(`https://planning-poker-241613.appspot.com/meetings/${meetingId}/tickets`, body),
  attendees: (meetingId) => getData(`https://planning-poker-241613.appspot.com/meetings/${meetingId}/attendees`),
  tickets: (meetingId) => getData(`http://localhost:9000/meetings/${meetingId}/tickets`)
};