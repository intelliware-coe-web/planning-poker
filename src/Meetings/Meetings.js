import React from 'react';
import {Link} from 'react-router-dom';
import './Meetings.css';

const meetings = ['First', 'Second', 'Third', 'Fourth', 'Fifth'];

export default function Meetings(props) {
    return (
        <>
        <div className="default-background">
            <div className="uk-container uk-text-center@m">
                <h1 className="uk-heading-divider uk-margin-top">Meetings</h1>
                <a href="#/">
                    <span uk-icon="icon: arrow-left; ratio: 3" className="uk-position-large uk-position-top-left"></span>
                </a>
                <div className="uk-align-center uk-width-1-2@m">                
                    { GetMeetingButtons() }
                    { GetAddMeetingButton() }
                </div>
            </div>
        </div>
        </>
    );
}

function GetMeetingButtons() {
    if (meetings.length === 0) {
        return <span>No meetings scheduled</span>
    }
    return meetings.map((meeting, index) => 
        <div key={index}>
                <Link to="/host">
                    <div className="uk-card uk-card-secondary uk-card-body meeting-card">
                        <h3 className="uk-card-title">{meeting}</h3>
                    </div>
                </Link>
            <br/>
        </div>
    );
}

function GetAddMeetingButton() {
    return (
        <Link to="/meeting/add">
            <div className="uk-card uk-card-secondary uk-card-body add-card">
                <h3 className="uk-card-title">+</h3>
            </div>
        </Link>
    );
}
