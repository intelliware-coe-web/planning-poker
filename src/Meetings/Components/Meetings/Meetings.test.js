import Meetings from "./Meetings";

it('Meetings renders correctly', () => {
    expect(Meetings()).toMatchSnapshot(); 
});