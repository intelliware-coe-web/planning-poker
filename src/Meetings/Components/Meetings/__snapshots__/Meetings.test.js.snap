// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetings renders correctly 1`] = `
<div
  className="default-background"
>
  <div
    className="uk-container uk-text-center@m"
  >
    <h1
      className="uk-heading-divider uk-margin-top"
    >
      Meetings
    </h1>
    <div
      className="uk-align-center uk-width-1-2@m"
    >
      <Memo(ConnectFunction) />
      <AddMeetingButton />
    </div>
  </div>
</div>
`;
